src/SourceGenerators/Uno.UI.Tasks/ResourcesGenerator/ResourcesGenerationTask.cs

```diff
@@ -171,6 +171,12 @@ private ITaskItem GenerateAndroidResources(string language, DateTime sourceLastW
 			var logicalTargetPath = Path.Combine(localizedDirectory, "strings.xml"); // this path is required by Xamarin
 			var actualTargetPath = Path.Combine(OutputPath, logicalTargetPath);
 
+			string fileName = Path.GetFileName(actualTargetPath);
+			if (fileName.Contains('-'))
+			{
+				this.Log().Error($"Invalid file name: must contain only [a-z0-9_.] {fileName} is invalid");
```

Comment:
I believe we should fail the compilation instead of logging an error. @jeromelaban what do you think?

Reply:
Also, for the different scales, images in UWP folders might be named "MyImage.scale-200.png", let's make sure this check doesn't flag those as invalid names as those are valid, they would be placed in Android "drawable-xhdpi\MyImage.png"

Reply:
My-Image.scale-200.png would be invalid.

Reply:
https://github.com/nventive/Uno.SourceGeneration#logging-to-build-output

Reply:
Oh that's nice! Would `logger.Error($"some error message");` make the compilation fail?

Reply:
Im not sure of how to use the logger given that I have no SourceGenerationContext